src/main/logutils.ts

```diff
@@ -22,66 +17,51 @@ let testRunning: boolean = false;
+ * will be emitted ('DataTimeout') to be able to clean up whatever was going on.
  */
-const interestingCombatLogEvents: InterestingCombatLogEventsType = {
-    'ARENA_MATCH_START': handleArenaStartLine,
-    'ARENA_MATCH_END': handleArenaStopLine,
-    'CHALLENGE_MODE_START': handleChallengeModeStartLine,
-    'CHALLENGE_MODE_END': handleChallengeModeEndLine,
-    'ENCOUNTER_START': handleEncounterStartLine,
-    'ENCOUNTER_END': handleEncounterStopLine,
-    'ZONE_CHANGE': handleZoneChange,
-    'COMBATANT_INFO': handleCombatantInfoLine,
-    'SPELL_AURA_APPLIED': handleSpellAuraAppliedLine,
-    'UNIT_DIED': handleUnitDiedLine,
-};
+const combatLogParser = new CombatLogParser({
+    dataTimeout: 2 * 60 * 1000
+});
 
 /**
- * A parsed line from the WoW combat log
- */
-class LogLine {
-    constructor (
-        // Timestamp in string format, as-is, from the log
-        // Example: '8/3 22:09:58.548'
-        public timestamp: string,
-
-        // Multi-dimensional array of arguments
-        // Example: 'ARENA_MATCH_START', '2547', '33', '2v2', '1'
-        public args: any[]
-    ) {}
+ * Log line handlers attached to their respective combat log event types.
+ * Any combat log event is valid here.
+ *
+ * Callback signature is:
+ *
+ * (line: LogLine, flavour: string)
+ *
+ * 'flavour' is the WoW flavour like 'wow_classic', 'wow' (retail)
+ */
+combatLogParser
+    .on('ARENA_MATCH_START', handleArenaStartLine)
+    .on('ARENA_MATCH_END', handleArenaStopLine)
+    .on('CHALLENGE_MODE_START', handleChallengeModeStartLine)
+    .on('CHALLENGE_MODE_END', handleChallengeModeEndLine)
+    .on('ENCOUNTER_START', handleEncounterStartLine)
+    .on('ENCOUNTER_END', handleEncounterStopLine)
+    .on('ZONE_CHANGE', handleZoneChange)
+    .on('COMBATANT_INFO', handleCombatantInfoLine)
+    .on('SPELL_AURA_APPLIED', handleSpellAuraAppliedLine)
+    .on('UNIT_DIED', handleUnitDiedLine)
+;
+
+// If we haven't received data in a while, we're probably AFK and should stop recording.
+combatLogParser.on('DataTimeout', (timeoutMs: number) => {
+    console.log(`[CombatLogParser] Haven't received data for combatlog in ${timeoutMs / 1000} seconds.`)
 
     /**
-     * Parse the timestamp from a log line and create a Date value from it
-     *
-     * Split the line by any delimiter that isn't a number
+     * End the current challenge mode dungeon and stop recording.
+     * We'll keep the video.
      */
-    date (): Date {
-        const timeParts = this.timestamp
-            .split(/[^0-9]/, 6)
-            .map(v => parseInt(v, 10))
-            .reverse();
-        const [msec, secs, mins, hours, day, month] = timeParts;
-        const dateObj = new Date();
-
-        if (day) dateObj.setDate(day);
-        if (month) dateObj.setMonth(month - 1);
-        dateObj.setHours(hours);
-        dateObj.setMinutes(mins);
-        dateObj.setSeconds(secs);
-        dateObj.setMilliseconds(msec);
-
-        return dateObj;
+    if (activeChallengeMode) {
```

Comment:
Do we want other content types in here? I think we certainly want BGs and possibly others. BGs are particularly hard to deal with as we rely on ZONE_CHANGE events only, and the timeout could be big value there.

Reply:
I don't know if we do - I just knew that Keystone Dungeons need this. I hope that those of you who do PvP can chime in with how we could use this to handle other content types here?

Reply:
For now can we just add BGs this? 

```
if (activeChallengeMode || isBG)
...
```